src/build/util.py

```diff
@@ -0,0 +1,64 @@
+# SPDX-License-Identifier: MIT
+
+import os
+import pathlib
+import sys
+import tempfile
+import typing
+
+from typing import Union
+
+import pep517
+
+import build
+import build.env
+
+
+if sys.version_info >= (3, 8):
+    import importlib.metadata as importlib_metadata
+else:
+    import importlib_metadata
+
+
+if typing.TYPE_CHECKING:
+    from importlib_metadata import PackageMetadata, SimplePath
+
+
+def _project_wheel_metadata(builder: build.ProjectBuilder) -> 'PackageMetadata':
+    with tempfile.TemporaryDirectory() as tmpdir:
+        path = pathlib.Path(builder.metadata_path(tmpdir))
+        if typing.TYPE_CHECKING:
+            cast_path = typing.cast(SimplePath, path)
```

Comment:
Why? `Path` should satisfy the `SimplePath` protocol.

Reply:
https://github.com/python/importlib_metadata/issues/334

Reply:
This is a case of me shooting myself in the foot :upside_down_face:. https://github.com/python/importlib_metadata/commit/fa20a0a330175502b61de89e5dd8e10f3f77ede7